src/firestore.js

```diff
@@ -402,20 +402,57 @@ function defaultHistory() {
   });
 }
 
+function hasDietaryRestrictions(userID)
+{
+  let docRef = db.collection("users").doc(userID).get();
+  const data = docRef.data();
+  return data["dietaryRestrictions"].length === 0
+}
+
 function getGroupInfo(groupID) {
   let docRef = db.collection("groups").doc(groupID).get();
-  // const data = docRef.data;
-  // data.id = groupID;
-  // return data;
+  const data = docRef.data();
+  const users = data['users']
+  const keywords = ""
+  const price = ""
+  const diet = []
+  for (let i = 0; i < users.length; i++)
+  {
+    keywords = keywords +  " " + users[i]["keywords"]
```

Comment:
Users is only a list of user ids. Iterating through it is fine, but the map you're looking for on line 421 is under the `data` key of the group.